provider/azure/credentials.go

```diff
@@ -90,9 +91,56 @@ func (c environProviderCredentials) CredentialSchemas() map[cloud.AuthType]cloud
 }
 
 // DetectCredentials is part of the environs.ProviderCredentials
-// interface.
+// interface. It attempts to detect subscription IDs from accounts
+// configured in the Azure CLI.
 func (c environProviderCredentials) DetectCredentials() (*cloud.CloudCredential, error) {
-	return nil, errors.NotFoundf("credentials")
+	// Attempt to get accounts from az.
+	accounts, err := c.azureCLI.ListAccounts()
+	if err != nil {
+		logger.Debugf("error getting accounts from az: %s", err)
+		return nil, errors.NotFoundf("credentials")
+	}
+	if len(accounts) < 1 {
+		return nil, errors.NotFoundf("credentials")
+	}
+	clouds, err := c.azureCLI.ListClouds()
+	if err != nil {
+		logger.Debugf("error getting clouds from az: %s", err)
+		return nil, errors.NotFoundf("credentials")
+	}
+	cloudMap := make(map[string]azurecli.Cloud, len(clouds))
+	for _, cloud := range clouds {
+		cloudMap[cloud.Name] = cloud
+	}
+	var defaultCredential string
+	authCredentials := make(map[string]cloud.Credential)
+	for i, acc := range accounts {
+		cloudInfo := cloudMap[acc.CloudName]
+		cred, err := c.accountCredential(acc, cloudInfo)
+		if err != nil {
+			logger.Debugf("cannot get credential for %s: %s", acc.Name, err)
+			if i == 0 {
```

Comment:
mainly because if it fails it's probably because get-access-token isn't implemented, so there seems little point in keeping going.